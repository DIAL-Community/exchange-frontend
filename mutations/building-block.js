import { gql } from '@apollo/client'

export const CREATE_BUILDING_BLOCK = gql`
  mutation CreateBuildingBlock(
    $name: String!
    $slug: String!
    $maturity: String!
    $imageFile: Upload
    $description: String!
    $specUrl: String
  ) {
    createBuildingBlock(
      name: $name
      slug: $slug
      maturity: $maturity
      imageFile: $imageFile
      description: $description
      specUrl: $specUrl
    ) {
      buildingBlock {
        slug
      }
      errors
    }
  }
`
