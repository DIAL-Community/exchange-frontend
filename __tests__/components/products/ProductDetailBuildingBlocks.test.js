import { fireEvent, screen } from '@testing-library/react'
import { mockRouterImplementation, mockSessionImplementation, render, waitForReactSelectToLoad } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { BUILDING_BLOCK_SEARCH_QUERY } from '../../../queries/building-block'
import ProductDetailBuildingBlocks from '../../../components/products/ProductDetailBuildingBlocks'
import { buildingBlocks } from './data/ProductDetailBuildingBlocks'
import { product } from './data/ProductForm'

// Mock next-router calls.
jest.mock('next/dist/client/router')
// Mock the next-auth's useSession.
jest.mock('next-auth/client')

describe('Unit tests for the ProductDetailBuildingBlocks component.', () => {
  const EDIT_BUTTON_TEST_ID = 'edit-button'
  const CANCEL_BUTTON_TEST_ID = 'cancel-button'
  const BUILDING_BLOCK_SEARCH_TEST_ID = 'building-block-search'
  const BUILDING_BLOCK_SEARCH_PLACEHOLDER = 'Type to search...'
  const BUILDING_BLOCK_SEARCH_OPTION_1_LABEL = 'Building Block 1'
  const BUILDING_BLOCK_SEARCH_OPTION_2_LABEL = 'Building Block 2'
  const PRODUCT_TEST_BUILDING_BLOCK_LABEL = 'Test Building Block'
  const PILL_TEST_ID = 'pill'
  const PILL_REMOVE_BUTTON_TEST_ID = 'remove-button'
  const mockBuildingBlocks = generateMockApolloData(BUILDING_BLOCK_SEARCH_QUERY, { search: '' }, null, buildingBlocks)

  beforeAll(() => {
    mockRouterImplementation()
    mockSessionImplementation()
  })

  test('Should match snapshot - without edit permission.', () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockBuildingBlocks]}>
        <ProductDetailBuildingBlocks
          canEdit={false}
          product={product}
        />
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should match snapshot - with edit permission.', () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockBuildingBlocks]}>
        <ProductDetailBuildingBlocks
          canEdit={true}
          product={product}
        />
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should match snapshot - with open editable section', async () => {
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockBuildingBlocks]}>
        <ProductDetailBuildingBlocks
          canEdit={true}
          product={product}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await screen.findByText(BUILDING_BLOCK_SEARCH_PLACEHOLDER)
    await waitForReactSelectToLoad(container)
    expect(container).toMatchSnapshot()
  })

  test('Should remove a pill', async () => {
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockBuildingBlocks]}>
        <ProductDetailBuildingBlocks
          canEdit={true}
          product={product}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await screen.findByText(BUILDING_BLOCK_SEARCH_PLACEHOLDER)
    await waitForReactSelectToLoad(container)
    fireEvent.click(getByTestId(PILL_REMOVE_BUTTON_TEST_ID))
    expect(screen.queryByTestId(PILL_TEST_ID)).toBeNull()
    expect(container).toMatchSnapshot()
  })

  test('Should add a pill and revert changes on "Cancel" button click', async () => {
    const { container, getByTestId, getByText } = render(
      <CustomMockedProvider mocks={[mockBuildingBlocks]}>
        <ProductDetailBuildingBlocks
          canEdit={true}
          product={product}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await screen.findByText(BUILDING_BLOCK_SEARCH_PLACEHOLDER)
    await waitForReactSelectToLoad(container)

    fireEvent.keyDown(getByTestId(BUILDING_BLOCK_SEARCH_TEST_ID).childNodes[1], { key: 'ArrowDown' })
    await screen.findByText(BUILDING_BLOCK_SEARCH_OPTION_1_LABEL)
    fireEvent.click(getByText(BUILDING_BLOCK_SEARCH_OPTION_1_LABEL))
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(2)

    fireEvent.keyDown(getByTestId(BUILDING_BLOCK_SEARCH_TEST_ID).childNodes[1], { key: 'ArrowDown' })
    await screen.findByText(BUILDING_BLOCK_SEARCH_OPTION_2_LABEL)
    fireEvent.click(getByText(BUILDING_BLOCK_SEARCH_OPTION_2_LABEL))
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(3)

    expect(container).toMatchSnapshot()

    fireEvent.click(getByTestId(CANCEL_BUTTON_TEST_ID))
    expect(screen.queryByText(PRODUCT_TEST_BUILDING_BLOCK_LABEL)).toBeInTheDocument()
    expect(screen.queryByText(BUILDING_BLOCK_SEARCH_OPTION_1_LABEL)).not.toBeInTheDocument()
    expect(screen.queryByText(BUILDING_BLOCK_SEARCH_OPTION_2_LABEL)).not.toBeInTheDocument()

    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await screen.findByText(BUILDING_BLOCK_SEARCH_PLACEHOLDER)

    expect(screen.queryByText(PRODUCT_TEST_BUILDING_BLOCK_LABEL)).toBeInTheDocument()
    expect(screen.queryByText(BUILDING_BLOCK_SEARCH_OPTION_1_LABEL)).not.toBeInTheDocument()
    expect(screen.queryByText(BUILDING_BLOCK_SEARCH_OPTION_2_LABEL)).not.toBeInTheDocument()
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(1)
  })
})
