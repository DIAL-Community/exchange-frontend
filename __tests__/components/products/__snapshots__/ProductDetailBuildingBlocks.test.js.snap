// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the ProductDetailBuildingBlocks component. Should add a pill and revert changes on "Cancel" button click 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-sapphire"
      >
        Building Blocks
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-stratos mb-3"
      >
        Assign
         
        Building Blocks
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
      >
        Mapping Status
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r4:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-qv4grc-control"
          >
            <div
              class="react-select__value-container react-select__value-container--has-value css-17t7gkf-ValueContainer"
            >
              <div
                class="react-select__single-value css-1dimb5e-singleValue"
              >
                Validated
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="building-block-search"
      >
        Search and Assign Building Blocks
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r5:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-qv4grc-control"
          >
            <div
              class="react-select__value-container css-17t7gkf-ValueContainer"
            >
              <div
                class="react-select__placeholder css-5rjclb-placeholder"
                id="react-select-:r5:-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-:r5:-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="async-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 0; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      >
        <div
          class="shadow-md flex gap-2 px-2 py-1 bg-white text-dial-stratos"
          data-testid="pill"
        >
          Test Building Block
          <svg
            class="cursor-pointer opacity-50"
            data-testid="remove-button"
            fill="currentColor"
            height="1.5em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </div>
        <div
          class="shadow-md flex gap-2 px-2 py-1 bg-white text-dial-stratos"
          data-testid="pill"
        >
          Building Block 1
          <svg
            class="cursor-pointer opacity-50"
            data-testid="remove-button"
            fill="currentColor"
            height="1.5em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </div>
        <div
          class="shadow-md flex gap-2 px-2 py-1 bg-white text-dial-stratos"
          data-testid="pill"
        >
          Building Block 2
          <svg
            class="cursor-pointer opacity-50"
            data-testid="remove-button"
            fill="currentColor"
            height="1.5em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </div>
      </div>
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the ProductDetailBuildingBlocks component. Should match snapshot - with edit permission. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-sapphire"
      >
        Building Blocks
      </div>
      <div
        class="gap-3"
      >
        <button
          class="mr-2 cursor-pointer bg-dial-iris-blue px-2 py-0.5 rounded text-white"
          data-testid="edit-button"
          type="button"
        >
          <svg
            class="inline pb-0.5"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 20h9"
            />
            <path
              d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
            />
          </svg>
          <span
            class="text-sm px-1"
          >
            Edit
          </span>
        </button>
      </div>
    </div>
    <div
      class="text-sm text-dial-gray-dark pb-2 highlight-link"
    >
      
    Note: As part of the 
      <a
        href="https://govstack.global"
      >
        GovStack Initiative
      </a>
      , DIAL is working
    to map digital platforms with key Building Block functionalities. Currently, all mappings between
    products and building blocks are "Beta" and are shown as illustrative examples of how these tools
    may support building block functionalities. As Building Block specifications are developed,
    products that are aligned with the specifications will be tagged as such.
  
    </div>
    <div
      class="grid grid-cols-1"
    >
      <a
        href="/building_blocks/test_building_block"
      >
        <div
          class="cursor-pointer hover:rounded-lg hover:shadow-lg border-3 border-transparent hover:border-dial-sunshine"
        >
          <div
            class="bg-white border border-dial-gray shadow-lg rounded-md"
          >
            <div
              class="flex flex-row flex-wrap gap-x-2 lg:gap-x-4 px-4 py-6"
            >
              <div
                class="w-10/12 lg:w-4/12 flex gap-2 my-auto text-dial-sapphire"
              >
                <div
                  class="block w-6 h-6 relative opacity-60"
                >
                  <img
                    alt="Logo for: Test Building Block"
                    class="object-contain"
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    sizes="100vw"
                    src="/_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=3840&q=75"
                    srcset="/_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=640&q=75 640w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=750&q=75 750w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=828&q=75 828w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=3840&q=75 3840w"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                  />
                </div>
                <div
                  class="w-full font-semibold"
                >
                  Test Building Block
                </div>
              </div>
              <div
                class="ml-auto font-semibold my-auto"
              >
                <div
                  class="text-dial-stratos opacity-70 text-xs font-semibold"
                >
                  BETA
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the ProductDetailBuildingBlocks component. Should match snapshot - with open editable section 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-sapphire"
      >
        Building Blocks
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-stratos mb-3"
      >
        Assign
         
        Building Blocks
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
      >
        Mapping Status
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r0:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-qv4grc-control"
          >
            <div
              class="react-select__value-container react-select__value-container--has-value css-17t7gkf-ValueContainer"
            >
              <div
                class="react-select__single-value css-1dimb5e-singleValue"
              >
                Validated
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="building-block-search"
      >
        Search and Assign Building Blocks
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r1:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-qv4grc-control"
          >
            <div
              class="react-select__value-container css-17t7gkf-ValueContainer"
            >
              <div
                class="react-select__placeholder css-5rjclb-placeholder"
                id="react-select-:r1:-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-:r1:-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="async-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      >
        <div
          class="shadow-md flex gap-2 px-2 py-1 bg-white text-dial-stratos"
          data-testid="pill"
        >
          Test Building Block
          <svg
            class="cursor-pointer opacity-50"
            data-testid="remove-button"
            fill="currentColor"
            height="1.5em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </div>
      </div>
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          disabled=""
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the ProductDetailBuildingBlocks component. Should match snapshot - without edit permission. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-sapphire"
      >
        Building Blocks
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="text-sm text-dial-gray-dark pb-2 highlight-link"
    >
      
    Note: As part of the 
      <a
        href="https://govstack.global"
      >
        GovStack Initiative
      </a>
      , DIAL is working
    to map digital platforms with key Building Block functionalities. Currently, all mappings between
    products and building blocks are "Beta" and are shown as illustrative examples of how these tools
    may support building block functionalities. As Building Block specifications are developed,
    products that are aligned with the specifications will be tagged as such.
  
    </div>
    <div
      class="grid grid-cols-1"
    >
      <a
        href="/building_blocks/test_building_block"
      >
        <div
          class="cursor-pointer hover:rounded-lg hover:shadow-lg border-3 border-transparent hover:border-dial-sunshine"
        >
          <div
            class="bg-white border border-dial-gray shadow-lg rounded-md"
          >
            <div
              class="flex flex-row flex-wrap gap-x-2 lg:gap-x-4 px-4 py-6"
            >
              <div
                class="w-10/12 lg:w-4/12 flex gap-2 my-auto text-dial-sapphire"
              >
                <div
                  class="block w-6 h-6 relative opacity-60"
                >
                  <img
                    alt="Logo for: Test Building Block"
                    class="object-contain"
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    sizes="100vw"
                    src="/_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=3840&q=75"
                    srcset="/_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=640&q=75 640w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=750&q=75 750w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=828&q=75 828w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2F%2Ftest-building-block.png&w=3840&q=75 3840w"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                  />
                </div>
                <div
                  class="w-full font-semibold"
                >
                  Test Building Block
                </div>
              </div>
              <div
                class="ml-auto font-semibold my-auto"
              >
                <div
                  class="text-dial-stratos opacity-70 text-xs font-semibold"
                >
                  BETA
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the ProductDetailBuildingBlocks component. Should remove a pill 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-sapphire"
      >
        Building Blocks
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-stratos mb-3"
      >
        Assign
         
        Building Blocks
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
      >
        Mapping Status
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r2:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-qv4grc-control"
          >
            <div
              class="react-select__value-container react-select__value-container--has-value css-17t7gkf-ValueContainer"
            >
              <div
                class="react-select__single-value css-1dimb5e-singleValue"
              >
                Validated
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="building-block-search"
      >
        Search and Assign Building Blocks
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r3:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-qv4grc-control"
          >
            <div
              class="react-select__value-container css-17t7gkf-ValueContainer"
            >
              <div
                class="react-select__placeholder css-5rjclb-placeholder"
                id="react-select-:r3:-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-:r3:-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="async-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      />
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;
